File: javascript/27_remove_element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
	var i = 0,
    	l = nums.length;
    var res = 0;
    for (;i < l; i ++) {
    	if (nums[i] !== val) {
    		nums[res++] = nums[i];
    	}
    }

    return res;
};	

File: python/27_remove_element.py
class Solution:
    # @param {integer[]} nums
    # @param {integer} val
    # @return {integer}
    def removeElement(self, nums, val):
    	res = 0
    	l = len(nums)
    	i = 0
    	while i < l:
    		if nums[i] != val:
    			nums[res] = nums[i]
    			res += 1
    		i += 1

    	print(nums)
    	return res

